File: integrations/kotlin/src/jsMain/kotlin/angular/ts/Injector.kt
package angular.ts

import angular.ts.generated.InjectorService as RawInjectorService

public class Injector internal constructor(
    internal val raw: RawInjectorService<Any?>,
) {
    public fun <T> get(token: Token<T>): T =
        token.fromJs(raw.get(token.name))

    public fun has(token: Token<*>): Boolean =
        raw.has(token.name)
}

File: integrations/scala/src/main/scala/angular/ts/Injector.scala
package angular.ts

final class Injector private[ts] (private val raw: RuntimeInjector):
  def get[A](token: Token[A]): A =
    raw.get(token.name).asInstanceOf[A]

object Injector:
  private[ts] def apply(raw: RuntimeInjector): Injector = new Injector(raw)
